compiler/scopelang/scope_to_dcalc.ml

```diff
@@ -655,11 +678,11 @@ let translate_rules
   let return_exp =
     Bindlib.box_apply
       (fun args ->
-        ETuple (args, Some sigma_return_struct_name), pos_mark pos_sigma)
+        ETuple (args, Some sigma_return_struct_name), mark_tany mark pos_sigma)
```

Comment:
So here the Scope -> Dcalc translation does not try to fill in all the types it introduces; rather it relies on a re-typing pass. Right ?

Reply:
Yes, it's a best-effort: at first I tried to introduce more "manually-computed" types but it was too tedious and time-consuming.

The good approach would be to use "smart constructors" to generate the new terms (the `Expr.make_*` functions). For this specific case I actually did add `Expr.make_tuple` later on, so that could be used here.

But in general, having the new terms with `TAny` and then re-typing to infer their type and checking consistency with both their parents and sub-terms doesn't seem like a bad approach, it has a small speed impact (re-typing is required) but offers the same guarantees. One limitation at the moment is that we don't distinguish between `typed expr` with full typing and `typed expr` that may contain holes ; this could be added at some point but for now it remains manageable.